examples/programmatic/llpr/llpr.py

```diff
@@ -0,0 +1,170 @@
+
+
+qm9_systems = read_systems("qm9_reduced_100.xyz", dtype=torch.float64)
+
+target_config = {
+    "energy": {
+        "quantity": "energy",
+        "read_from": "ethanol_reduced_100.xyz",
+        "file_format": ".xyz",
+        "key": "energy",
+        "unit": "kcal/mol",
+        "forces": False,
+        "stress": False,
+        "virial": False,
+    },
+}
+targets, _ = read_targets(target_config, dtype=torch.float64)
+
+requested_neighbor_lists = model.requested_neighbor_lists()
+qm9_systems = [
+    get_system_with_neighbor_lists(system, requested_neighbor_lists)
+    for system in qm9_systems
+]
+dataset = Dataset({"system": qm9_systems, **targets})
+
+# We also load a single ethanol molecule on which we will compute properties.
+# This system is loaded without targets, as we are only interested in the LPR
+# values.
+ethanol_system = read_systems("ethanol_reduced_100.xyz", dtype=torch.float64)[0]
+ethanol_system = get_system_with_neighbor_lists(
+    ethanol_system, requested_neighbor_lists
+)
+
+# %%
+#
+# The dataset is fully compatible with torch. For example, be used to create
+# a DataLoader object.
+
+from metatrain.utils.data import collate_fn  # noqa: E402
+
+
+dataloader = torch.utils.data.DataLoader(
+    dataset,
+    batch_size=10,
+    shuffle=False,
+    collate_fn=collate_fn,
+)
+
+
+# %%
+#
+# We now wrap the model in a LLPRModel object, which will allows us to compute
+# prediction rigidity metrics, which are useful for uncertainty quantification
+# and model introspection.
+
+from metatensor.torch.atomistic import (  # noqa: E402
+    MetatensorAtomisticModel,
+    ModelMetadata,
+)
+
+from metatrain.utils.llpr import LLPRModel  # noqa: E402
+
+
+llpr_model = LLPRModel(model)
+llpr_model.compute_covariance(dataloader)
+llpr_model.compute_inverse_covariance(regularizer=1e-4)
+
+exported_model = MetatensorAtomisticModel(
+    llpr_model.eval(),
+    ModelMetadata(),
+    llpr_model.capabilities,
+)
+
+# %%
+#
+# We can now use the model to compute the LPR for every atom in the ethanol molecule.
+# To do so, we create a ModelEvaluationOptions object, which is used to request
+# specific outputs from the model. In this case, we request the uncertainty in the
+# atomic energy predictions.
+
+from metatensor.torch.atomistic import ModelEvaluationOptions, ModelOutput  # noqa: E402
+
+
+evaluation_options = ModelEvaluationOptions(
+    length_unit="angstrom",
+    outputs={
+        "mtt::aux::energy_uncertainty": ModelOutput(per_atom=True),
+        # you can request any output from the model here, for example:
+        # "energy": ModelOutput(per_atom=True),
+        # "mtt::aux::last_layer_features": ModelOutput(per_atom=True),
+    },
+    selected_atoms=None,
+)
+
+outputs = exported_model([ethanol_system], evaluation_options, check_consistency=True)
+lpr = outputs["mtt::aux::energy_uncertainty"].block().values.detach().cpu().numpy()
```

Comment:
Can we also expose here how the (total) PR can be deduced? Even if it's just a scalar number, should be fine.